src/rust/catnapw/futures/connect.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//==============================================================================
+// Imports
+//==============================================================================
+
+use socket2::{
+    SockAddr,
+    Socket,
+};
+use windows::Win32::Networking::WinSock::{
+    WSAEALREADY,
+    WSAEINPROGRESS,
+    WSAEWOULDBLOCK,
+};
+
+use crate::runtime::{
+    fail::Fail,
+    QDesc,
+};
+use ::std::{
+    future::Future,
+    pin::Pin,
+    task::{
+        Context,
+        Poll,
+    },
+};
+use std::{
+    cell::RefCell,
+    rc::Rc,
+};
```

Comment:
Please, preappend `::` to crate-external imports:

```
use ::socket2::Socket;
use ::std::{..}
```

This avoids name clash.